tiny_dnn/core/kernels/fully_connected_op_internal.h

```diff
@@ -32,24 +32,28 @@ namespace tiny_dnn {
 namespace kernels {
 
 inline void
-fully_connected_op_internal(const tensor_t&     in_data,
-                            const vec_t&        W,
-                            const vec_t&        bias,
-                            tensor_t&           out_data,
-                            const fully_params& params,
-                            const bool          layer_parallelize) {
-    for_i(layer_parallelize, in_data.size(), [&](int sample) {
-        const vec_t& in = in_data[sample];
-        vec_t& out = out_data[sample];
+fully_connected_op_internal(const Tensor<float_t, 2>& in_data,
+                            const Tensor<float_t, 2>& weights,
+                            const Tensor<float_t, 2>& bias,
+                            Tensor<float_t, 2>*       out_data,
+                            const bool             parallelize) {
+    auto in_shape  = in_data.shape();
+    auto out_shape = out_data->shape();
 
-        for (serial_size_t i = 0; i < params.out_size_; i++) {
-            out[i] = float_t(0);
-            for (serial_size_t c = 0; c < params.in_size_; c++) {
-                out[i] += W[c * params.out_size_ + i] * in[c];
+    const float_t*  W = &*weights.host_begin();
```

Comment:
we could but it might happens that there's no bias and then an exception is raised. Probably it's easier to provide a method that return a pointer to beginning of the vector